gamemodes/terrortown/gamemode/shared/sh_walkspeed.lua

```diff
@@ -0,0 +1,95 @@
+local plymeta = assert(FindMetaTable("Player"), "FAILED TO FIND ENTITY TABLE")
+
+SPEED = SPEED or {}
+
+if SERVER then
+	---
+	-- Sets the initial @{Player} speed multiplier, called in @{GM:TTTBeginRound} after (dumb) players
+	-- are respawned.
+	-- @realm server
+	-- @internal
+	function SPEED:InitPlayerSpeedMultiplier()
+		local plys = player.GetAll()
+
+		for i = 1, #plys do
+			local ply = plys[i]
+
+			ply:SetSpeedMultiplier(1.0)
+		end
+	end
+
+	---
+	-- Sets the speed multiplier to an absolute value
+	-- @param number value The new value
+	-- @realm server
+	function plymeta:SetSpeedMultiplier(value)
+		self:TTT2NETSetFloat("player_speed_multiplier", value)
+	end
+
+	---
+	-- Multiplies the @{Player}s current speed modifier with the given value
+	-- @param number value The multiplier
+	-- @realm server
+	function plymeta:GiveSpeedMultiplier(value)
+		self:TTT2NETSetFloat("player_speed_multiplier", self:TTT2NETGetFloat("player_speed_multiplier", 1.0) * value)
+	end
+
+	---
+	-- Devides the @{Player}s current speed modifier with the given value
+	-- @param number value The multiplier
+	-- @realm server
+	function plymeta:RemoveSpeedMultiplier(value)
+		self:TTT2NETSetFloat("player_speed_multiplier", self:TTT2NETGetFloat("player_speed_multiplier", 1.0) / value)
+	end
+end
+
+if CLIENT then
+	---
+	-- Initializes the speed system once the game is ready.
+	-- It is called in @{GM:Initialize}.
+	-- @realm client
+	function SPEED:Initialize()
+		STATUS:RegisterStatus("ttt_walkspeed_status_good", {
+			hud = {
+				Material("vgui/ttt/perks/hud_speedrun.png")
+			},
+			type = "good",
+			DrawInfo = function()
+				return math.Round(LocalPlayer():GetSpeedMultiplier(), 1)
+			end
+		})
+
+		STATUS:RegisterStatus("ttt_walkspeed_status_bad", {
+			hud = {
+				Material("vgui/ttt/perks/hud_speedrun.png")
+			},
+			type = "bad",
+			DrawInfo = function()
+				return math.Round(LocalPlayer():GetSpeedMultiplier(), 1)
+			end
+		})
+
+		TTT2NET:OnUpdate("players", function(oldval, newval, reversePath)
```

Comment:
Using only the "players" path, will lead to an update call for any player whose multiplier has changed and not only for your own player. You will need to append the Steamid64 to the path or use https://github.com/TTT-2/TTT2/blob/4947aa62ef89616b4e7350c195013e885ee8b200/gamemodes/terrortown/gamemode/client/cl_network_sync.lua#L370
which will take care of prefixing the correct path.

Reply:
But then I need to register a function for every player which is rather problematic because I have to track latejoiners etc.

Reply:
Please review the function of this code, which is only for the `LocalPlayer()` or am I wrong?

Reply:
Yes, you're right. But It did not work for me with `TTT2NET:OnUpdateOnPlayer`, maybe I did something else wrong.